addon/components/bs4/bs-form/element/label.js

```diff
@@ -1,9 +1,65 @@
+import Ember from 'ember';
 import FormElementLabel from 'ember-bootstrap/components/base/bs-form/element/label';
 
+const {
+  computed
+} = Ember;
+
 export default FormElementLabel.extend({
   tagName: 'label',
 
-  classNames: ['form-control-label'],
-  classNameBindings: ['invisibleLabel:sr-only', 'labelClass'],
-  attributeBindings: ['formElementId:for']
+  classNames: [],
+  classNameBindings: ['invisibleLabel:sr-only', 'isHorizontal:col-form-label', 'isCheckbox:form-check-label', 'labelClass'],
+  attributeBindings: ['formElementId:for'],
+
+  /**
+   * The form layout used for the markup generation (see http://getbootstrap.com/css/#forms):
+   *
+   * * 'horizontal'
+   * * 'vertical'
+   * * 'inline'
+   *
+   * Defaults to the parent `form`'s `formLayout` property.
+   *
+   * @property formLayout
+   * @type string
+   * @default 'vertical'
+   * @public
+   */
+  formLayout: 'vertical',
+
+  /**
+   * The type of the control widget.
+   * Supported types:
+   *
+   * * 'text'
+   * * 'checkbox'
+   * * 'textarea'
+   * * any other type will use an input tag with the `controlType` value as the type attribute (for e.g. HTML5 input
+   * types like 'email'), and the same layout as the 'text' type
+   *
+   * @property controlType
+   * @type string
+   * @default 'text'
+   * @public
+   */
+  controlType: 'text',
```

Comment:
I would prefer these two additional properties (`controlType` and `formLayout` to live in `base/bs-form/element/label`, so we can keep using only `common/bs-form/element.hbs` for bs3 and bs4. This would hide the fact that the bs4 label renders differently based on these properties as an implementation detail of that component, and we don't have to care otherwise.